### PR TITLE
Allow copying password to clipboard.

### PR DESCRIPTION
**Pull request by nevack**
- Fix layout wrapping with flex groups.
- Factor out CRUD to separate module.
- Add `onSubmit` desctucturing helper.

### CODE REVIEW

File: public/js/utils.js [hunk 1]
```diff
@@ -4,3 +4,29 @@ export function onClick(element, action) {
...
+        new FormData(form).forEach((value, key) => {
+            values[key] = value;
+        });
+        action(values)
+    })
+}
+
+export let copyHelper = (name = "copy-helper") => {
+    return {
+        inject: () => `<input type="text" id="${name}" style="position: absolute; left: -100vw">`,
```
Comment (nevack) [inline public/js/utils.js:21]: [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) requires Permissions API with manifest.
This is the most simple workaround.

**Comment by nevack** [comment]
image

image

File: public/js/utils.js [hunk 2]
```diff
@@ -4,3 +4,29 @@ export function onClick(element, action) {
...
+        copy: (text) => {
+            console.log("Copy requested for #" + name);
+            const injected = document.getElementById(name);
```
Comment (gmltA) [inline public/js/utils.js:24]: I guess you can introduce a guard here for a case when no input element was injected previously.

File: public/js/data.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return passQuery.docs.map(query => {
+        const password = query.data();
+        password.id = query.id;
+        return password;
+    });
+}
+
+export function getFavicon(website) {
+    const hostname = (new URL(website)).hostname || "example.com";
+    return "https://s2.googleusercontent.com/s2/favicons?domain=" + hostname;
```
Comment (gmltA) [inline public/js/data.js:27]: 👍

**Pull request merged**